### PR TITLE
OO bib entry 

### PR DESCRIPTION
**Pull request by womullan**
not ready yet

### CODE REVIEW

**Comment by womullan** [comment]
@jonathansick now this is used in the generatebib.  For a first step I think its good. Pybtex runs over the entire thing but I guess in another improvement i could parse the static file and sort the entire bib instead of just concatinating - though concat also keeps clean the generated from the static..

**Comment by womullan** [comment]
@jonathansick ok for this ??   dont want to get too out of date with main

File: bin/bibtools.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+"""This class represents a bibtex bibentry.
+it can construct from a string and output to the bib file as needed.
+it is comparable hence sortable.
```
Comment (jonathansick) [inline bin/bibtools.py:5]: I don't quite follow how `BibEntry` comparisons work? Wouldn't it need `__eq__`, `__le__`, etc [rich comparison](https://docs.python.org/3/reference/datamodel.html#object.__lt__) methods?

File: bin/bibtools.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "SMTN": "Simulations Team Technical Note",
+    "SQR": "SQuaRE Technical Note",
+    "DMTR": "Data Management Test Report",
+    "LDM": "Data Management Controlled Document",
+}
+
+
+class BibEntry:
+    """A class to represent a BibEntry
+    # not all entries have all fields
```
Comment (jonathansick) [inline bin/bibtools.py:25]: Maybe a typo in the docstring with the line comment syntax?

File: bin/bibtools.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        self.year = year
+        self.type = "@DocuShare"
+        if "" == note and handle:
+            prefix, _ = self.handle.split("-", 1)
+            series = TN_SERIES.get(prefix, "")
+            self.note = f"{{Vera C. Rubin Observatory {series}}}"
+
+    def write_latex_bibentry(self, fd=sys.stdout):
+        """ Write a bibentry for document info passed.
+        This assumes its an lsst.io entry and constructs the url accordingly.
```
Comment (jonathansick) [inline bin/bibtools.py:44]: Is this comment about URLs from an earlier version of the code? `BibEntry` doesn't seem to be doing any URL formatting.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -18,13 +18,19 @@ jobs:
...
 
       - name: Run flake8
         run: flake8
 
       - name: Run pybtex
         run: make test-pybtex
 
+      - name: Run pytest
+        env:
+          PYTHONPATH: bin
```
Comment (jonathansick) [inline .github/workflows/ci.yaml:31]: I see this as a bit of a code smell for packaging issues, though it's not worth changing now. If we get a chance we can come back and put in a more standard infrastructure around lsst-texmf's Python tools.

**Review by jonathansick — APPROVED** [review]
Seems ok.

File: bin/bibtools.py (hunk 1 above)
Comment (womullan) [reply bin/bibtools.py:5]: yes i have not done that yet .. future effort

Comment (womullan) [reply bin/bibtools.py:3]: comparators added

**Pull request merged**